framework/src/partitioner/PetscExternalPartitioner.C

```diff
@@ -110,21 +139,30 @@ PetscExternalPartitioner::_do_partition(MeshBase & mesh, const unsigned int n_pa
   nj = 0;
   for (auto & row : _dual_graph)
   {
-    if (_apply_side_weight)
-      neighbor = 0;
+    auto & elem = _local_id_to_elem[local_elem_id];
```

Comment:
Potential RHS insert! Put a mooseAssert here. Also you don't need to capture a pointer by reference - this actually confused me.

Reply:
I see a bunch of PETSc calls above (not part of this PR) without return code checking. Those are allocs too - you probably should add them.